src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -984,6 +985,10 @@ trait Contexts { self: Analyzer =>
 
     def isNameInScope(name: Name) = lookupSymbol(name, _ => true).isSuccess
 
+    def isPlainClassInJavaUnit(sym: Symbol): Boolean =
+      unit.isJava && sym.isJavaDefined && sym.isClass && !sym.isModuleClass && !sym.isPackageClass
```

Comment:
Why exclude `sym.isModuleClass`? I'm trying to get this scenario to work (using scaladoc to more eagerly force infos):
```
public class A {
    public static class T {}
}
public class B extends A {
    public static class U extends T { }
//                                  ^
// error: not found: type T
}
```